File: 27-05-2024_Special_Array_With_X_Elements_Greater_Than_or_Equal_X.java
class Solution {
    public int specialArray(int[] nums) {
        Arrays.sort(nums);
        int n = nums.length;
        int left = 0, right = n;

        while (left <= right) {
            int mid = left + (right - left) / 2;
            int count = 0;
            for (int num : nums) {
                if (num >= mid) {
                    count++;
                }
            }

            if (count == mid) {
                return mid;
            } else if (count > mid) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }

        return -1;
    }
}

File: 27-05-2024_Special_Array_With_X_Elements_Greater_Than_or_Equal_X.py
from typing import List

class Solution:
    def specialArray(self, nums: List[int]) -> int:
        nums.sort()
        n = len(nums)
        left, right = 0, n

        while left <= right:
            mid = left + (right - left) // 2
            count = sum(num >= mid for num in nums)

            if count == mid:
                return mid
            elif count > mid:
                left = mid + 1
            else:
                right = mid - 1

        return -1
